From c27d5ade2b051257c8106c7e1a686e2fda3fe49c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:21 -0400
Subject: [tools] perf/lib/traceevent: Fix memory leak in pretty_print()

Message-id: <1410181752-25631-137-git-send-email-jolsa@redhat.com>
Patchwork-id: 91842
O-Subject: [PATCH RHEL7.1 BZ1134356 136/307] tools lib traceevent: Fix memory leak in pretty_print()
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit de04f8657de9d3351a2d5880f1f7080b23b798cf
Author: Steven Rostedt <rostedt@goodmis.org>
Date: Tue Apr 22 19:23:30 2014 -0400

description
===========
Commit 12e55569a244 "tools lib traceevent: Use helper trace-seq in print
functions like kernel does" added a extra trace_seq helper to process
string arguments like the kernel does it. But the difference between the
kernel and the userspace library is that the kernel's trace_seq structure
has a static allocated buffer. The userspace one has a dynamically
allocated one. It requires a trace_seq_destroy(), otherwise it produces
a nasty memory leak.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-parse.c b/tools/lib/traceevent/event-parse.c
index baec7d8..b83184f 100644
--- a/tools/lib/traceevent/event-parse.c
+++ b/tools/lib/traceevent/event-parse.c
@@ -4344,6 +4344,7 @@ static void pretty_print(struct trace_seq *s, void *data, int size, struct event
            format, len_arg, arg);
     trace_seq_terminate(&p);
     trace_seq_puts(s, p.buffer);
+    trace_seq_destroy(&p);
     arg = arg->next;
     break;
    default:
-- 
1.7.1